web/app/scripts/export/export-controller.js

```diff
@@ -0,0 +1,98 @@
+     * Exports are tied to ctl.tilekey, which comes via the directive from MapController.
+     * Changing filters changes tilekey which resets the download.
+     */
+
+    /* ngInject */
+    function ExportController($rootScope, $scope, $interval, Exports, InitialState) {
+        var ctl = this;
+        var pollingInterval;
+        var POLLING_INTERVAL_MS = 1500;
+        var MAX_POLLING_TIME_S = 100;
+
+        InitialState.ready().then(initialize);
+
+        function initialize() {
+            ctl.isOpen = false;
+            ctl.pending = false;
+            ctl.downloadURL = null;
+            ctl.error = null;
+
+            ctl.toggle = toggle;
+            ctl.exportCSV = exportCSV;
+        }
+
+        function toggle() {
+            ctl.isOpen = !ctl.isOpen;
+            if (ctl.isOpen) {
+                $rootScope.$broadcast('driver.export.open');
+            }
+        }
+
+        function exportCSV() {
+            ctl.error = null;
+            Exports.create({ tilekey: ctl.tilekey },
+                function (result) {
+                    pollForDownload(result.taskid);
+                }, function () {
+                    ctl.error = 'Error initializing export.';
+                }
+            );
+        }
+
+        function pollForDownload(taskID) {
+            ctl.pending = true;
+            pollingInterval = $interval(function () {
+                    Exports.get({ id: taskID }).$promise.then(function (response) {
+                        switch (response.status) {
+                            case 'PENDING':
+                            case 'STARTED':
+                                break;
+                            case 'FAILURE':
+                                ctl.error = response.error;
+                                cancelPolling();
+                                break;
+                            case 'SUCCESS':
+                                ctl.downloadURL = response.result;
+                                cancelPolling();
+                                break;
+                        }
+                    });
+                },
+                POLLING_INTERVAL_MS,
+                MAX_POLLING_TIME_S * 1000 / POLLING_INTERVAL_MS
+            );
+            // The interval's promise resolves if it hits the limit without being cancelled
+            pollingInterval.then(function () {
+                cancelPolling();
+                ctl.error = 'Export request timed out.';
+            });
+        }
+
+        function cancelPolling() {
+            ctl.pending = false;
+            $interval.cancel(pollingInterval);
+        }
+
+        // Clears any in-progress or completed download, so they can do a new one if they want.
+        function reset() {
+            cancelPolling();
+            ctl.error = null;
+            ctl.downloadURL = null;
+        }
+
+        $scope.$watch(function () { return ctl.tilekey; }, reset);
```

Comment:
I see where you're going with this but I'm wondering whether it might get confusing for users; I could see someone wanting to kick off a CSV export, click around to set up a new set of filters while the first one is exporting, and then start another export job once the first one finishes, and I think it might be surprising if that caused their pending CSV export to disappear.